Source/WebCore/platform/mediastream/RealtimeMediaSource.h

```diff
@@ -416,6 +421,25 @@ struct PhotoCapabilitiesOrError {
     String errorMessage;
 };
 
+struct PhotoSettingsOrError {
```

Comment:
With a NativePromise you wouldn't need to create this object.

Simply create a NativePromise<PhotoSettings, String> and return this.

Reply:
This is the pattern we have used so far in media capture. It may make sense to change it, but I am going to follow the pattern for consistency here and we can change them all at once.